test/spec/proxyBenchmark.js

```diff
@@ -58,69 +43,153 @@ function generateBigObjectFixture(carsSize) {
+      const obj = generateBigObjectFixture(1);
+      modifyObj(obj);
+    });
+  }
+
+  {
+    suite.add(`${suiteName} (JSONPatcherProxy)`, function() {
+      const obj = generateBigObjectFixture(1);
+      const jsonPatcherProxy = new JSONPatcherProxy(obj);
+      const observedObj = jsonPatcherProxy.observe(true);
+      const patches = jsonPatcherProxy.generate();
+      modifyObj(observedObj);
+    });
+  }
+  
+  {
+    suite.add(`${suiteName} (fast-json-patch)`, function() {
+      const obj = generateBigObjectFixture(1);
+      const observer = jsonpatch.observe(obj);
+      modifyObj(obj);
+      jsonpatch.generate(observer);
+    });
+  }
+}
+
+/* ============================= */
+
+{
+  const suiteName = 'Observe and generate';
+  
+  {
+    suite.add(`${suiteName} (noop)`, function() {
+      const obj = generateBigObjectFixture(100);
+      modifyObj(obj);
+    });
+  }
+
+  {
+    suite.add(`${suiteName} (JSONPatcherProxy)`, function() {
+      const obj = generateBigObjectFixture(100);
+      const jsonPatcherProxy = new JSONPatcherProxy(obj);
+      const observedObj = jsonPatcherProxy.observe(true);
+      const patches = jsonPatcherProxy.generate();
+      modifyObj(observedObj);
+    });
+  }
+  
+  {
+    suite.add(`${suiteName} (fast-json-patch)`, function() {
+      const obj = generateBigObjectFixture(100);
+      const observer = jsonpatch.observe(obj);
+      modifyObj(obj);
+      jsonpatch.generate(observer);
+    });
+  }
+}
+
+/* ============================= */
+
+{
+  const suiteName = 'Primitive mutation';
+
+  {
+    const obj = generateBigObjectFixture(100);
+    
+    suite.add(`${suiteName} (noop)`, function() {
+      obj.cars[50].name = reverseString(obj.cars[50].name);
+    });
+  }
+
+  {
+    const obj = generateBigObjectFixture(100);
+    const jsonPatcherProxy = new JSONPatcherProxy(obj);
+    const observedObj = jsonPatcherProxy.observe(true);
+    
+    suite.add(`${suiteName} (JSONPatcherProxy)`, function() {
+      observedObj.cars[50].name = reverseString(observedObj.cars[50].name);
+    });
+  }
+
+  {
+    const obj = generateBigObjectFixture(100);
+    const observer = jsonpatch.observe(obj);
+    
+    suite.add(`${suiteName} (fast-json-patch)`, function() {
+      obj.cars[50].name = reverseString(obj.cars[50].name);
+      jsonpatch.generate(observer);
+    });
+  }
+}
+
+/* ============================= */
+
+{
+  const suiteName = 'Complex mutation';
+
+  {
+    const obj = generateBigObjectFixture(100);
+  
+    suite.add(`${suiteName} (noop)`, function() {
+      const item = obj.cars.shift();
+      obj.cars.push(item);
+    });
+  }
+
+  {
+    const obj = generateBigObjectFixture(100);
+    const jsonPatcherProxy = new JSONPatcherProxy(obj);
+    const observedObj = jsonPatcherProxy.observe(true);
+  
+    suite.add(`${suiteName} (JSONPatcherProxy)`, function() {
+      const item = observedObj.cars.shift();
+      observedObj.cars.push(item);
```

Comment:
There is no `generate` call

Reply:
I didn't change this. It is how it works on `master`. But I think you are right, I will change it. 

@alshakero do you remember why did you chose to call `generate` here not later?

Reply:
When it comes to JSONPatcherProxy, `generate` call isn't needed here, or anywhere to be honest. All the heavy-lifting occurs synchronously upon the mutation itself. I think it's safe to remove all `generate` calls.

Reply:
Since the heavy-lifting occurs synchrously, `generate` has low cost but makes the test symmetrical with the test done using `fast-json-patch`. Hence, I think it is good to keep `generate` in both places